core/templates/domain/collection/collection-update.service.ts

```diff
@@ -19,243 +19,266 @@
+   * undo/redo service.
+   */
+  setCollectionObjective(collection: Collection, objective: string): void {
+    let oldObjective = collection.getObjective();
+    this._applyPropertyChange(
+      collection, CollectionDomainConstants.COLLECTION_PROPERTY_OBJECTIVE,
+      objective, oldObjective, (changeDict, collection) => {
+        // Apply.
+        let objective = this._getNewPropertyValueFromChangeDict(changeDict);
+        collection.setObjective(objective);
+      }, (changeDict, collection) => {
+        // Undo.
+        collection.setObjective(oldObjective);
+      });
+  }
 
-      /**
-       * Changes the language code of a collection and records the change in
-       * the undo/redo service.
-       */
-      setCollectionLanguageCode: function(collection, languageCode) {
-        var oldLanguageCode = angular.copy(collection.getLanguageCode());
-        _applyPropertyChange(
-          collection, COLLECTION_PROPERTY_LANGUAGE_CODE, languageCode,
-          oldLanguageCode,
-          function(changeDict, collection) {
-            // Apply.
-            var languageCode = _getNewPropertyValueFromChangeDict(changeDict);
-            collection.setLanguageCode(languageCode);
-          }, function(changeDict, collection) {
-            // Undo.
-            collection.setLanguageCode(oldLanguageCode);
-          });
-      },
+  /**
+   * Changes the language code of a collection and records the change in
+   * the undo/redo service.
+   */
+  setCollectionLanguageCode(
+      collection: Collection,
+      languageCode: string): void {
+    let oldLanguageCode = collection.getLanguageCode();
+    this._applyPropertyChange(
+      collection, CollectionDomainConstants.COLLECTION_PROPERTY_LANGUAGE_CODE,
+      languageCode, oldLanguageCode, (changeDict, collection) => {
+        // Apply.
+        let languageCode = this._getNewPropertyValueFromChangeDict(changeDict);
+        collection.setLanguageCode(languageCode);
+      }, (changeDict, collection) => {
+        // Undo.
+        collection.setLanguageCode(oldLanguageCode);
+      });
+  }
 
-      /**
-       * Changes the tags of a collection and records the change in
-       * the undo/redo service.
-       */
-      setCollectionTags: function(collection, tags) {
-        var oldTags = angular.copy(collection.getTags());
-        _applyPropertyChange(
-          collection, COLLECTION_PROPERTY_TAGS, tags, oldTags,
-          function(changeDict, collection) {
-            // Apply.
-            var tags = _getNewPropertyValueFromChangeDict(changeDict);
-            collection.setTags(tags);
-          }, function(changeDict, collection) {
-            // Undo.
-            collection.setTags(oldTags);
-          });
-      },
+  /**
+   * Changes the tags of a collection and records the change in
+   * the undo/redo service.
+   */
+  setCollectionTags(collection: Collection, tags: string[]): void {
+    let oldTags = collection.getTags();
```

Comment:
Can this be changed to const, since it is not going to be updated again in this function.

Reply:
updated.